### ISSUE TITLE
LOG_LEVEL not respected in dotenv

### ISSUE THREAD

**Issue by jnordberg** [vapor/vapor#2429]
Setting `LOG_LEVEL=debug` in a dotfile has no effect

### Steps to reproduce

In `.env.development`:

```
LOG_LEVEL=debug
```

```shell
$ vapor run
```

### Expected behavior

The debug logs should be printed to console just like when running: `$ LOG_LEVEL=debug vapor run`

### Actual behavior

The log level is at `info`

### Environment

```
vapor --version
Error: Error: Unknown command `--version`
```

Looks like this github issue template is outdated :)
* Vapor Framework version: 4.14.0
* Vapor Toolbox version: ?
* OS version: 10.15.5 (19F101)

**Comment by 0xTim** [comment]
The reason for this is that the logging environment is bootstrapped before dot env files are read (and before the application which reads those files exists). I think you'll need to pass it in via an environment variable

**Comment by tanner0101** [comment]
Yeah the solution to this is not very straightforward unfortunately. Right now Application lets the user control LoggingSystem.bootstrap completely. It expects that is setup before it is initialized. However, `.env` file reading doesn't happen until after Application has initialized. 

I've been meaning to explore an option for having Application control LoggingSystem.bootstrap for you. In that situation, we could first load `.env` before configuring logging.

**Comment by MrLotU** [comment]
@tanner0101 With my draft of using @main for Vapor you could do this quite simply. Check https://github.com/vapor/vapor/tree/LotU-atmain for a WIP.

In this setup you would end up with something like this:
```swift
@main
struct MyApp: Application {
     static func configure(...) { }
     static func routes(...) { }
     static func setupLogging(...) { }
}
```

`Application` in it's `static func main()` would first load the `.env`, and then call the above functions, probably in order of `setupLogging`, `configure` and `routes`

**Comment by 0xTim** [comment]
This should be solved by the current template https://github.com/vapor/template/blob/main/Sources/App/entrypoint.swift
